rpc/v8/block.go

```diff
@@ -158,6 +165,47 @@ func (h *Handler) BlockWithReceipts(id BlockID) (*BlockWithReceipts, *jsonrpc.Er
 	}, nil
 }
 
+func (h *Handler) BlockWithTxHashesAndReceipts(id BlockID) (*BlockWithTxHashesAndReceipts, *jsonrpc.Error) {
```

Comment:
Here you should return `BlockWithTxHashesAndReceipts` instead of `*Block...AndReceipts`. The reason is that the latter always forces a heap allocation which is sub-optimal. Read the recent group interactions in slack to understand a bit more